src/azure-cli/azure/cli/command_modules/network/custom.py

```diff
@@ -6861,304 +6865,330 @@ def list_traffic_manager_endpoints(cmd, resource_group_name, profile_name, endpo
+            arg_group="Subnet",
+            help="Name or ID of a network security group (NSG).",
+            fmt=AAZResourceIdArgFormat(
+                template="/subscriptions/{subscription}/resourceGroups/{resource_group}/providers/Microsoft.Network"
+                         "/networkSecurityGroups/{}",
+            ),
+        )
+        # set default value
+        args_schema.address_prefixes._default = ["10.0.0.0/16"]
+        # filter arguments
+        args_schema.extended_location._registered = False
+        return args_schema
 
-    if network_security_group:
-        instance.network_security_group = NetworkSecurityGroup(id=network_security_group)
-    elif network_security_group == '':  # clear it
-        instance.network_security_group = None
+    def pre_operations(self):
+        from azure.cli.core.aaz import has_value
+        args = self.ctx.args
+        if has_value(args.edge_zone):
+            args.extended_location.name = args.edge_zone
+            args.extended_location.type = "EdgeZone"
+
+        if has_value(args.subnet_name):
+            subnet = {"name": args.subnet_name}
+            if not has_value(args.subnet_prefixes):
+                # set default value
+                address, bit_mask = str(args.address_prefixes[0]).split("/")
+                subnet_mask = 24 if int(bit_mask) < 24 else bit_mask
+                subnet["address_prefix"] = f"{address}/{subnet_mask}"
+            elif len(args.subnet_prefixes) == 1:
+                subnet["address_prefix"] = args.subnet_prefixes[0]
+            else:
+                subnet["address_prefixes"] = args.subnet_prefixes
+            if has_value(args.subnet_nsg):
+                subnet["network_security_group"] = {"id": args.subnet_nsg}
+            args.subnets = [subnet]
 
-    _set_route_table(network_client_factory(cmd.cli_ctx), resource_group_name, route_table, instance)
+    def _output(self, *args, **kwargs):
+        result = self.deserialize_output(self.ctx.vars.instance, client_flatten=True)
+        return {"newVNet": result}
 
-    if service_endpoints == ['']:
-        instance.service_endpoints = None
-    elif service_endpoints:
-        instance.service_endpoints = []
-        for service in service_endpoints:
-            instance.service_endpoints.append(ServiceEndpoint(service=service))
 
-    if service_endpoint_policy == '':
-        instance.service_endpoint_policies = None
-    elif service_endpoint_policy:
-        instance.service_endpoint_policies = []
-        for policy in service_endpoint_policy:
-            instance.service_endpoint_policies.append(SubResource(id=policy))
+class VNetUpdate(_VNetUpdate):
+    def post_instance_update(self, instance):
+        if instance.properties.dhcp_options.dns_servers == [""]:
+            instance.properties.dhcp_options.dns_servers = None
+        if str(instance.properties.ddos_protection_plan).split("/")[-1] == "":
+            instance.properties.ddos_protection_plan = None
```

Comment:
If we want to deprecate this behavior, recommend handle this in pre_operations with warnings.
Recommend using customize `AAZResourceIdArgFormat`.